### PR TITLE
Add stacking switcher in area charts

### PR DESCRIPTION
**Pull request by DaryaLari**

### CODE REVIEW

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-10648-1818/wizard-1818/ru/">wizard (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-10648-1818/wizard-1818/en/">wizard (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-10648-1818/wizard-1818/ru/">wizard (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-10648-1818/wizard-1818/en/">wizard (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>Errors occurred when merging changes from your branch with the Weblate branch.</summary>

```
Rebasing (1/1)
Auto-merging src/i18n-keysets/sql/en.json
CONFLICT (add/add): Merge conflict in src/i18n-keysets/sql/en.json
Auto-merging src/i18n-keysets/sql/ru.json
CONFLICT (add/add): Merge conflict in src/i18n-keysets/sql/ru.json
Auto-merging src/i18n-keysets/wizard/en.json
CONFLICT (add/add): Merge conflict in src/i18n-keysets/wizard/en.json
Auto-merging src/i18n-keysets/wizard/ru.json
CONFLICT (add/add): Merge conflict in src/i18n-keysets/wizard/ru.json
Auto-merging src/server/modes/charts/plugins/datalens/highcharts/highcharts.ts
CONFLICT (add/add): Merge conflict in src/server/modes/charts/plugins/datalens/highcharts/highcharts.ts
Auto-merging src/ui/registry/units/common/functions-map.ts
CONFLICT (add/add): Merge conflict in src/ui/registry/units/common/functions-map.ts
Auto-merging src/ui/registry/units/common/register.tsx
CONFLICT (add/add): Merge conflict in src/ui/registry/units/common/register.tsx
Auto-merging src/ui/store/actions/openDialogTypes.ts
CONFLICT (add/add): Merge conflict in src/ui/store/actions/openDialogTypes.ts
Auto-merging src/ui/units/dash/components/ParamsSettings/helpers.ts
CONFLICT (add/add): Merge conflict in src/ui/units/dash/components/ParamsSettings/helpers.ts
Auto-merging src/ui/units/dash/components/ParamsSettings/test/validateParamTitle.test.ts
CONFLICT (add/add): Merge conflict in src/ui/units/dash/components/ParamsSettings/test/validateParamTitle.test.ts
Auto-merging src/ui/units/dash/containers/Dialogs/Dialogs.js
CONFLICT (add/add): Merge conflict in src/ui/units/dash/containers/Dialogs/Dialogs.js
error: could not apply 29fdece7b... Add stacking switcher
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply 29fdece7b... Add stacking switcher
 (1)
```

</details>

**Resolve conflicts according to instructions**
1. Switch to the current branch associated with this pull request.
```
git checkout CHARTS-10648
```
2. Add Weblate as remote. And remove previous upstream with the same name:
```
git remote remove weblate 2> /dev/null
git remote add weblate https://weblate.gravity-ui.com/git/datalens/charts-10648-1818/chartkit-1818/
git -c http.sslVerify=false remote update weblate
```
3. Merge Weblate changes:
```
git merge weblate/CHARTS-10648
```
4. Resolve conflicts:
```
edit ...
git add ...
git commit
```
5. Push changes to upstream repository, Weblate will fetch merge from there:
```
git push origin
```

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>The following components have not been translated</summary>
<p><a href="https://weblate.gravity-ui.com/projects/datalens/charts-10648-1818/sql-1818/ru/">sql (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-10648-1818/sql-1818/en/">sql (en)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-10648-1818/wizard-1818/ru/">wizard (ru)</a><br><a href="https://weblate.gravity-ui.com/projects/datalens/charts-10648-1818/wizard-1818/en/">wizard (en)</a></p>
</details>

Wait for the reviewers to check your changes in Weblate and try running github action again.

File: src/shared/modules/config/wizard/v10/mapV10ConfigToV11.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import {ChartsConfigVersion, IndicatorTitleMode} from '../../../../types';
 export const mapV10ConfigToV11 = (config: V10ChartsConfig): V11ChartsConfig => {
     let extraSettings = config.extraSettings as V11CommonSharedExtraSettings;
     if (config.visualization?.id === WizardVisualizationId.Metric) {
-        extraSettings = {...config.extraSettings};
+        extraSettings = {...config.extraSettings, stacking: 'on'};
```
Comment (kuzmadom) [inline src/shared/modules/config/wizard/v10/mapV10ConfigToV11.ts:12]: Why was it necessary to add this default value for `WizardVisualizationId.Metric`?

**Comment by robot-charts** [comment]
**i18n-check**
<details>
<summary>Errors occurred when merging changes from your branch with the Weblate branch.</summary>

```
warning: skipped previously applied commit 6228d48ca
warning: skipped previously applied commit 31aa6b251
warning: skipped previously applied commit e501b0a53
warning: skipped previously applied commit 54c4e0e49
hint: use --reapply-cherry-picks to include skipped commits
hint: Disable this message with "git config advice.skippedCherryPicks false"
Rebasing (1/1)
Auto-merging src/i18n-keysets/sql/en.json
CONFLICT (add/add): Merge conflict in src/i18n-keysets/sql/en.json
Auto-merging src/i18n-keysets/sql/ru.json
CONFLICT (add/add): Merge conflict in src/i18n-keysets/sql/ru.json
Auto-merging src/i18n-keysets/wizard/en.json
CONFLICT (add/add): Merge conflict in src/i18n-keysets/wizard/en.json
Auto-merging src/i18n-keysets/wizard/ru.json
CONFLICT (add/add): Merge conflict in src/i18n-keysets/wizard/ru.json
Auto-merging src/server/modes/charts/plugins/datalens/highcharts/highcharts.ts
CONFLICT (add/add): Merge conflict in src/server/modes/charts/plugins/datalens/highcharts/highcharts.ts
Auto-merging src/server/modes/charts/plugins/ql/yagr.ts
CONFLICT (add/add): Merge conflict in src/server/modes/charts/plugins/ql/yagr.ts
Auto-merging src/shared/modules/config/wizard/v10/mapV10ConfigToV11.ts
CONFLICT (add/add): Merge conflict in src/shared/modules/config/wizard/v10/mapV10ConfigToV11.ts
Auto-merging src/ui/components/WorkbookNavigationMinimal/WorkbookNavigationMinimal.scss
CONFLICT (add/add): Merge conflict in src/ui/components/WorkbookNavigationMinimal/WorkbookNavigationMinimal.scss
Auto-merging src/ui/registry/units/common/functions-map.ts
CONFLICT (add/add): Merge conflict in src/ui/registry/units/common/functions-map.ts
Auto-merging src/ui/registry/units/common/register.tsx
CONFLICT (add/add): Merge conflict in src/ui/registry/units/common/register.tsx
Auto-merging src/ui/store/actions/openDialogTypes.ts
CONFLICT (add/add): Merge conflict in src/ui/store/actions/openDialogTypes.ts
Auto-merging src/ui/units/dash/components/ParamsSettings/helpers.ts
CONFLICT (add/add): Merge conflict in src/ui/units/dash/components/ParamsSettings/helpers.ts
Auto-merging src/ui/units/dash/components/ParamsSettings/test/validateParamTitle.test.ts
CONFLICT (add/add): Merge conflict in src/ui/units/dash/components/ParamsSettings/test/validateParamTitle.test.ts
Auto-merging src/ui/units/dash/containers/Dialogs/Dialogs.js
CONFLICT (add/add): Merge conflict in src/ui/units/dash/containers/Dialogs/Dialogs.js
Auto-merging src/ui/units/wizard/components/Dialogs/DialogPointsSize.tsx
CONFLICT (add/add): Merge conflict in src/ui/units/wizard/components/Dialogs/DialogPointsSize.tsx
error: could not apply 29fdece7b... Add stacking switcher
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply 29fdece7b... Add stacking switcher
 (1)
```

</details>

**Resolve conflicts according to instructions**
1. Switch to the current branch associated with this pull request.
```
git checkout CHARTS-10648
```
2. Add Weblate as remote. And remove previous upstream with the same name:
```
git remote remove weblate 2> /dev/null
git remote add weblate https://weblate.gravity-ui.com/git/datalens/charts-10648-1818/chartkit-1818/
git -c http.sslVerify=false remote update weblate
```
3. Merge Weblate changes:
```
git merge weblate/CHARTS-10648
```
4. Resolve conflicts:
```
edit ...
git add ...
git commit
```
5. Push changes to upstream repository, Weblate will fetch merge from there:
```
git push origin
```

**Review by kuzmadom — APPROVED** [review]

File: src/i18n-keysets/sql/en.json [hunk 2]
```diff
@@ -40,7 +40,7 @@
   "label_run": "Run",
   "label_select-connection": "Select",
   "label_select-visualization": "Select visualization",
-  "label_visualization-area": "Stacked area chart",
+  "label_visualization-area": "Area chart",
   "label_visualization-area-100p": "Normalized stacked area chart\n",
```
[the remark this answers is not in the document]
Comment (DaryaLari) [reply src/i18n-keysets/sql/en.json:44]: We've discussed in pc, that it is not critical

File: src/shared/modules/config/wizard/v10/mapV10ConfigToV11.ts (hunk 1 above)
Comment (DaryaLari) [reply src/shared/modules/config/wizard/v10/mapV10ConfigToV11.ts]: I've removed it

File: src/server/modes/charts/plugins/datalens/highcharts/highcharts.ts [hunk 3]
```diff
@@ -111,6 +111,12 @@ export const buildHighchartsConfigPrivate = (args: {
         });
     });
 
+    if (shared.visualization.id === 'area') {
```
[the remark this answers is not in the document]
Comment (DaryaLari) [reply src/server/modes/charts/plugins/datalens/highcharts/highcharts.ts:114]: I'll fix `visualization.id` usage from strings to enums in this file in separate pr

**Pull request merged**